src/modengine/ext/mod_loader/wwise_file_overrides.cpp

```diff
@@ -0,0 +1,120 @@
+#include <filesystem>
+
+#include "wwise_file_overrides.h"
+#include "archive_file_overrides.h"
+
+namespace modengine::ext {
+
+// This hook the IAkFileLocationResolver::open() method. It takes in a path 
+// and a so-called openMode. This openMode parameter is of type AkOpenMode 
+// which is an enum with 4 states: Read, Write, WriteOverwrite and ReadWrite.
+// Passing in any of these 4 invariants will cause this function to yield a 
+// FileOperator used for sourcing the file bytes. FromSoftware added another 
+// possible state to AKOpenMode, decimal 9. which will make this fn yield an 
+// EBLFileOperator, this implementation does all its fetching from the BDTs.
+// EBLFileOperator does not use the usual virtual path lookup that ME2 already 
+// hooks. Hence we need this hook here.
+//
+// In order to selectively make it read from disk again this hook checks if an 
+// override file exists and sets the openMode back from 9 to Read and replaces
+// the virtual path parameter string with an absolute path.
+//
+// This is a messy one though:
+// Figuring out if there is an override isn't straight forward. The hooked 
+// function gets invoked with a virtual path string ex: `sd:/50846376.wem`. 
+// Wwise uses subdirectories for localized content, meaning that the "simple"
+// WEM example makes Wwise look in `/50846376.wem` but also in 
+// `enus/50846376.wem` (or `ja/50846376.wem` for AC6 which has `ja` as an extra 
+// locale for some audio and BNKs). To make matters even worse, Elden Ring 
+// specifically sorts the WEMs into a subdir (`wem/`) and then another subdir 
+// based on the  first two digits of the WEM. So above example will spawn 
+// lookups in  `/wem/50/50846376.wem` and `enus/wem/50/50846376.wem`. In order 
+// to figure out if there is an override we will need to look in multiple 
+// directories per request. Luckily, aside from boot, this routine is called
+// quite infrequently.
+//
+// Also, worth pointing out that not all paths passed to this hook will have 
+// the `sd:/` prefix. So we cannot get away with the usual prefix / rewrite 
+// trick and will have to allocate a completely new string.
+
+namespace fs = std::filesystem;
+
+using namespace spdlog;
+
+std::optional<fs::path> find_override_file(const fs::path game_path)
```

Comment:
This is identical to the code in `archive_file_overrides.cpp`, right? If so I'd just move it to `mod_loader_extension.cpp` and share the code between them to prevent losing some debugging time to slight drift in the future.

Reply:
As per discussed in DMs I've removed the one in the wwise specific part and put a forward decl in the archive_file_overrides since the `wwise_file_overrides.cpp` -> `archive_file_overrides.h` dependency was already there

Should be resolved per b34fa3049ed37442a17e7a8b53989f97cf3f6c24